lms/product/canvas/_plugin/grouping.py

```diff
@@ -18,10 +18,36 @@ class CanvasGroupingPlugin(GroupingServicePlugin):
 
     group_type = Grouping.Type.CANVAS_GROUP
     sections_type = Grouping.Type.CANVAS_SECTION
+    deep_linking = True
 
-    def __init__(self, canvas_api):
+    def __init__(self, context, request, canvas_api):
+        self._context = context
+        self._request = request
         self._canvas_api = canvas_api
 
+    def get_grouping_type(self, svc) -> Grouping.Type:
+        """
+        Return the type of grouping used in this launch.
+
+        Grouping types describe how the course members are divided.
+        If neither of the LMS grouping features are used "COURSE" is the default.
+        """
+
+        if (
```

Comment:
I'd move this conditional to `sections_enabled` and this method can go away as I mention below.

I reckon that way what grouping type is being selected is more clear in all LMSs.